ui/raidboss/data/07-dt/trial/doomtrain.ts

```diff
@@ -0,0 +1,554 @@
+      condition: Conditions.targetIsYou(),
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Doomtrain Turret Side',
+      type: 'StartsUsing',
+      netRegex: { id: ['B23D'], capture: true },
+      suppressSeconds: 1,
+      infoText: (_data, matches, output) =>
+        output.avoid!({
+          dir: output[parseFloat(matches.x) < arenas[1].x ? 'west' : 'east']!(),
+        }),
+      outputStrings: {
+        avoid: {
+          en: 'Avoid ${dir} turrets',
+        },
+        east: Outputs.east,
+        west: Outputs.west,
+      },
+    },
+    {
+      id: 'Doomtrain Add Actor Finder',
+      type: 'ActorMove',
+      netRegex: { moveType: '0096' },
+      suppressSeconds: 9999,
+      run: (data, matches) => {
+        data.addTrainId = matches.id;
+      },
+    },
+    {
+      id: 'Doomtrain Add Train Direction Predictor',
+      type: 'HeadMarker',
+      netRegex: { id: '0282', data0: '1[0-9A-F]{7}', capture: true },
+      condition: (data, matches) => data.me === data.party?.idToName_?.[matches.data0],
+      durationSeconds: 7.6,
+      countdownSeconds: 7.6,
+      alertText: (data, matches, output) => {
+        const actor = data.actorPositions[data.addTrainId];
+        if (actor === undefined)
+          return;
+
+        let addCleaveDir = Math.atan2(actor.x - arenas.add.x, actor.y - arenas.add.y);
+
+        // Rough estimates, as I don't have a 0ms difference case to use to get exact numbers
+        // like I did for extreme
+        const slowMoveBase = 2.022;
+        const slowMoveDelta = 0.000367;
+
+        const deltaMs = new Date(matches.timestamp).getTime() - actor.time;
+
+        addCleaveDir -= slowMoveBase + (slowMoveDelta * deltaMs);
+
+        if (addCleaveDir < -Math.PI) {
+          addCleaveDir += Math.PI * 2;
+        }
+
+        const dirNum = Directions.hdgTo16DirNum(addCleaveDir);
+        const addTrainDir = dirNum !== undefined
+          ? Directions.output16Dir[dirNum] ?? 'unknown'
+          : 'unknown';
+
+        return output.text!({
+          dir: output[addTrainDir]!(),
+        });
+      },
+      outputStrings: {
+        ...Directions.outputStrings16Dir,
+        text: {
+          en: 'Tank cleave on YOU from ${dir}',
+        },
+      },
+    },
+    {
+      id: 'Doomtrain Derailment Siege',
+      type: 'StartsUsing',
+      netRegex: { id: 'B250', capture: false },
+      durationSeconds: 19.7,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Tower x3 => Next Platform',
+          de: 'Turm x3 => Nächste Plattform',
+          cn: '踩塔 x3 => 下一节车厢',
+        },
+      },
+    },
+    {
+      id: 'Doomtrain Headlight',
+      type: 'StartsUsing',
+      netRegex: { id: 'B244', capture: false },
+      durationSeconds: 6.7,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Down',
```

Comment:
I don't think we need it for now, but if Square continues with verticality mechanics like this we might want to add canned outputs for this in the future.